drivers/serial/serial_esp32_usb.c

```diff
@@ -71,27 +70,12 @@ static int serial_esp32_usb_err_check(const struct device *dev)
 	return 0;
 }
 
-#ifdef CONFIG_UART_USE_RUNTIME_CONFIGURE
-static int serial_esp32_usb_config_get(const struct device *dev, struct uart_config *cfg)
-{
-	return -ENOTSUP;
-}
-#endif /* CONFIG_UART_USE_RUNTIME_CONFIGURE */
-
-static int serial_esp32_usb_configure(const struct device *dev, const struct uart_config *cfg)
+static int serial_esp32_usb_init(const struct device *dev)
```

Comment:
To avoid any confusion, this driver does not use any part of Zephyr's USB support and should not have _usb_ in the name or in any designations. Fix it everywhere.

Reply:
It is named `USB Serial/JTAG Controller (USB_SERIAL_JTAG)` everywhere in the ESP32 documentation (see e.g. [ESP-IDF docs](https://docs.espressif.com/projects/esp-idf/en/latest/esp32c3/api-guides/usb-serial-jtag-console.html) and [reference manual](https://www.espressif.com/sites/default/files/documentation/esp32-c3_technical_reference_manual_en.pdf)) and it would be confusing to use a name that does not contain USB even though that's what this interface is.

I have deliberately put this driver under `serial` and not `USB`, which makes it clear enough from my perspective that it does not use the Zephyr USB stack.

This PR cleans up / fixes a driver that has already been merged and does not introduce the naming you are complaining about. If you want the name to be changed, please open a dedicated issue, but don't block this PR.

@carlescufi what is your opinion on this topic?

Reply:
> This PR cleans up / fixes a driver that has already been merged and does not introduce the naming you are complaining about. If you want the name to be changed, please open a dedicated issue, but don't block this PR.

You claim to clean it up because your PR #44045 was not accurate, so clean it up properly.

Reply:
Looks like we have different opinions regarding "proper" naming. I will wait for further feedback from other reviewers and change the driver name in a follow-up PR if that's desired by the majority.

Reply:
> Looks like we have different opinions regarding "proper" naming. I will wait for further feedback from other reviewers and change the driver name in a follow-up PR if that's desired by the majority.

It not just about name, all this is altogether not correct, like commit title [drivers: serial: esp32_usb: Add USB serial / JTAG interface](https://github.com/zephyrproject-rtos/zephyr/pull/44045/commits/23bdd715a645e9d72c73f56815c7adfad0a7eee2), it did not add USB serial or JTAG interface but a driver that uses build-in serial - USB adapter. Also Kconfig help says "Enable the ESP32 USB serial driver integrated in the JTAG interface." and there is jtag in the function identifiers, but it seems to be independent of JTAG and do not use it at all. More likely there is a USB - JTAG adapter, but what does it have to do with serial driver?

Reply:
@jfischer-no I get the point. If majority or TSC recommends to remove all usb/jtag calls from source files (not low-level calls like `usb_serial_jtag_ll_write_txfifo()`), I suggest to rename .c, h. and functions to `uart_esp32_builtin`. However, my opinion would be to maintain the `usb` as is to make very clear this serial interface is built on top of internal usb stack of ESP32-C3 SoC.

Reply:
I think that naming can be adjusted later if controversial, this PR is addressing other issues that are not related with the name, and so I don't think it's a reason to block it.

Reply:
> I'm absolutely fine with **constructive** criticism and I agree that the Kconfig description could be improved to make the details of this special type of interface more clear.

Well then improve it in this PR, what is preventing you to push a commit?